--- a/3rd_party/include/opentracing/variant/variant.hpp
+++ b/3rd_party/include/opentracing/variant/variant.hpp
@@ -167,7 +167,7 @@
 template <typename F, typename V, typename Enable = void>
 struct result_of_unary_visit
 {
-    using type = typename std::result_of<F(V&)>::type;
+    using type = typename std::invoke_result<F, V&>::type;
 };
 
 template <typename F, typename V>
@@ -179,7 +179,7 @@
 template <typename F, typename V, typename Enable = void>
 struct result_of_binary_visit
 {
-    using type = typename std::result_of<F(V&, V&)>::type;
+    using type = typename std::invoke_result<F, V&, V&>::type;
 };
 
 template <typename F, typename V>
